plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -543,17 +545,31 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         for (ConnectorExpression expression : expressions) {
             if (expression instanceof Call) {
                 Call call = (Call) expression;
-                // TODO Support ESCAPE character when it's pushed down by the engine
-                if (new FunctionName("$like_pattern").equals(call.getFunctionName()) && call.getArguments().size() == 2 &&
+                if (new FunctionName("$like_pattern").equals(call.getFunctionName()) &&
                         call.getArguments().get(0) instanceof Variable && call.getArguments().get(1) instanceof Constant) {
+                    String escape;
+
+                    switch (call.getArguments().size()) {
+                        case 2:
+                            escape = "\\";
+                            break;
+                        case 3:
+                            verify(call.getArguments().get(2) instanceof Constant, "Escape is not a constant");
```

Comment:
Who does guarantee that it's a constant?

Reply:
Done